ci/codecov.sh

```diff
@@ -91,7 +99,7 @@ elif [[ "$1" == "upload" ]]; then
     fi
 
     chmod +x codecov
-    ./codecov --verbose --nonZero ${CODECOV_NAME:+--name "$CODECOV_NAME"}
+    ./codecov --verbose --nonZero ${CODECOV_NAME:+--name "$CODECOV_NAME"} --file $(pwd)/coverage.info
```

Comment:
This change is either not required or incomplete:
- codecov searches for coverage files by default and finds a coverage.info in the current dir, so this can be removed
- Alternatively the file search can be disabled with `-X search`

Please choose one of those as otherwise the change causes more work to be done by the uploader

Reply:
I will try it again with the base directory and if it fails I will point you to the results in uuid.

Reply:
I'd be very much interested in that issue as I helped getting the file search to work which used to pick up to much but after my changes all my tests showed it working. You are right though: Unless `--directory` is passed it searches the root folder: https://github.com/codecov/uploader/blob/7a9a54e245e629c9a1c8891352cf6941ae4d2496/src/index.ts#L180

However: I don't see why this is an issue: We change back to that: https://github.com/boostorg/boost-ci/pull/135/files#diff-3b16f9fb8cf94a073f54db8870ba233ecba9779950aa103b7f8b80f2a256c932R53-R54

So that file must be in the folder codecov searches. If it does not or ignores something it shouldn't we can workaround it but I'd like to file an issue, so I'd need a log or something.

If it really is an issue: I'm fine with `--file` AND `-X search` which I even find preferable except for the verbosity.

Reply:
Ok, seen the issue on drone. The new uploader has a bug: https://github.com/codecov/uploader/issues/614

So `./codecov --verbose --nonZero ${CODECOV_NAME:+--name "$CODECOV_NAME"} --file $(pwd)/coverage.info --X search` should be the best.

Reply:
Without --file:

```
+ ./codecov --verbose --nonZero --name 'Github Actions'
[2022-02-04T16:32:58.144Z] ['verbose'] Start of uploader: 1643992378144...
[2022-02-04T16:32:58.145Z] ['info'] 
     _____          _
    / ____|        | |
   | |     ___   __| | ___  ___ _____   __
   | |    / _ \ / _` |/ _ \/ __/ _ \ \ / /
   | |___| (_) | (_| |  __/ (_| (_) \ V /
    \_____\___/ \__,_|\___|\___\___/ \_/

  Codecov report uploader 0.1.16
[2022-02-04T16:32:58.151Z] ['info'] => Project root located at: /home/runner/work/uuid/uuid
[2022-02-04T16:32:58.154Z] ['info'] -> No token specified or token is empty
[2022-02-04T16:32:58.154Z] ['verbose'] Start of network processing...
[2022-02-04T16:32:58.154Z] ['verbose'] Searching for files in /home/runner/work/uuid/uuid
[2022-02-04T16:32:58.158Z] ['info'] Searching for coverage files...
[2022-02-04T16:32:58.195Z] ['info'] Warning: Some files located via search were excluded from upload.
[2022-02-04T16:32:58.195Z] ['info'] If Codecov did not locate your files, please review https://docs.codecov.com/docs/supported-report-formats
[2022-02-04T16:32:58.195Z] ['error'] There was an error running the uploader: No coverage files located, please try use `-f`, or change the project root with `-R`
[2022-02-04T16:32:58.195Z] ['verbose'] The error stack is: Error: No coverage files located, please try use `-f`, or change the project root with `-R`
    at main (/snapshot/repo/dist/src/index.js)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
[2022-02-04T16:32:58.195Z] ['verbose'] End of uploader: 51 milliseconds
Error: Process completed with exit code 255.
```

Reply:
Using "--directory" also fails the same way.

Re: "In the other PR you updated the used GCC" I decided to remove as many differences as possible so I could get something committed that I can use to fix coverage in uuid.

Reply:
Here's the output when using lcov 1.15 and --base-directory is present:

```
+ lcov --gcov-tool=gcov-8 --rc lcov_branch_coverage=1 --base-directory /home/runner/work/uuid/boost-root/libs/uuid --directory /home/runner/work/uuid/boost-root --capture --output-file all.info
Capturing coverage data from /home/runner/work/uuid/boost-root
Found gcov version: 8.4.0
Using intermediate gcov format
Scanning /home/runner/work/uuid/boost-root for .gcda files ...
Found 108 data files in /home/runner/work/uuid/boost-root
Processing chrono/build/gcc-8/debug/cxxstd-11-iso/link-static/threading-multi/visibility-hidden/chrono.gcda
geninfo: WARNING: could not open /home/runner/work/uuid/boost-root/libs/uuid/boost/chrono/duration.hpp
geninfo: WARNING: could not open /home/runner/work/uuid/boost-root/libs/uuid/boost/chrono/detail/inlined/posix/chrono.hpp
geninfo: WARNING: could not open /home/runner/work/uuid/boost-root/libs/uuid/boost/system/detail/interop_category.hpp
geninfo: WARNING: could not open /home/runner/work/uuid/boost-root/libs/uuid/boost/system/system_error.hpp
geninfo: WARNING: could not open /home/runner/work/uuid/boost-root/libs/uuid/boost/system/detail/error_condition.hpp
geninfo: WARNING: could not open /home/runner/work/uuid/boost-root/libs/uuid/boost/system/detail/append_int.hpp
geninfo: WARNING: could not open /home/runner/work/uuid/boost-root/libs/uuid/boost/system/detail/throws.hpp
geninfo: WARNING: could not open /home/runner/work/uuid/boost-root/libs/uuid/boost/system/detail/system_category_impl.hpp
geninfo: WARNING: could not open /home/runner/work/uuid/boost-root/libs/uuid/boost/system/detail/error_category.hpp
geninfo: WARNING: could not open /home/runner/work/uuid/boost-root/libs/uuid/boost/exception/exception.hpp
geninfo: WARNING: could not open /home/runner/work/uuid/boost-root/libs/uuid/boost/throw_exception.hpp
geninfo: WARNING: could not open /home/runner/work/uuid/boost-root/libs/uuid/boost/system/detail/error_code.hpp
geninfo: WARNING: could not open /home/runner/work/uuid/boost-root/libs/uuid/boost/system/detail/error_category_impl.hpp
geninfo: WARNING: could not open /home/runner/work/uuid/boost-root/libs/uuid/boost/system/detail/generic_category_message.hpp
geninfo: WARNING: could not open /home/runner/work/uuid/boost-root/libs/uuid/boost/system/detail/snprintf.hpp
geninfo: WARNING: could not open /home/runner/work/uuid/boost-root/libs/uuid/boost/system/detail/std_category.hpp
geninfo: WARNING: could not open /home/runner/work/uuid/boost-root/libs/uuid/boost/system/detail/system_category.hpp
geninfo: WARNING: could not open /home/runner/work/uuid/boost-root/libs/uuid/boost/assert/source_location.hpp
geninfo: WARNING: could not open /home/runner/work/uuid/boost-root/libs/uuid/boost/system/detail/system_category_message.hpp
geninfo: WARNING: could not open /home/runner/work/uuid/boost-root/libs/uuid/boost/chrono/time_point.hpp
geninfo: WARNING: could not open /home/runner/work/uuid/boost-root/libs/uuid/boost/system/detail/generic_category.hpp
geninfo: WARNING: could not open /home/runner/work/uuid/boost-root/libs/uuid/boost/chrono/detail/system.hpp
geninfo: WARNING: some exclusion markers may be ignored
```

This is why I was trying to remove base-directory.

Reply:
But we're still not done.  The .codecov.yml in develop is not getting picked up.  Dammit, because it won't pick it up from the pull request OR from the target branch - it has to come from the main branch, I have to get .codecov.yml changes into master to see if it actually works or not.